triton-vm/src/table/program_table.rs

```diff
@@ -116,17 +118,29 @@ impl ExtProgramTable {
         vec![]
     }
 
-    fn ext_transition_constraints(
-        _challenges: &ProgramTableChallenges,
-    ) -> Vec<MPolynomial<XFieldElement>> {
+    pub fn ext_transition_constraints_as_circuits() -> Vec<MPolCircuit<ProgramTableChallenges>> {
+        let mut circuit_builder = MPolCircuitBuilder::default();
         let variables: Vec<MPolynomial<XFieldElement>> = MPolynomial::variables(2 * FULL_WIDTH);
+        let addr = circuit_builder.no_challenge(variables[usize::from(Address)].clone());
+        let addr_next =
+            circuit_builder.no_challenge(variables[FULL_WIDTH + usize::from(Address)].clone());
+        let one: MPolCircuitRef<ProgramTableChallenges> =
+            circuit_builder.no_challenge(MPolynomial::from_constant(1.into(), 2 * FULL_WIDTH));
+        let address_increases_by_one = addr_next - (addr + one);
 
-        let addr = variables[usize::from(Address)].clone();
-        let addr_next = variables[FULL_WIDTH + usize::from(Address)].clone();
-        let one = MPolynomial::<XFieldElement>::from_constant(1.into(), 2 * FULL_WIDTH);
+        vec![address_increases_by_one.consume()]
+    }
 
-        let address_increases_by_one = addr_next - (addr + one);
-        vec![address_increases_by_one]
+    fn ext_transition_constraints(
+        challenges: &ProgramTableChallenges,
+    ) -> Vec<MPolynomial<XFieldElement>> {
+        let circuits = Self::ext_transition_constraints_as_circuits();
+        let mut ret: Vec<MPolynomial<XFieldElement>> = vec![];
+        for circuit in circuits {
+            ret.push(circuit.partial_evaluate(challenges));
+        }
+
+        ret
```

Comment:
This is the second time I see this exact code, but where the challenges have a different type. Would it make sense to implement `ext_transition_constraints` generically on the trait?

Use `Vec::with_capacity(circuits.len())` or simply:

```suggestion
        Self::ext_transition_constraints_as_circuits()
            .iter()
            .map(|circuit| circuit.partial_evaluate(challenges))
            .collect()
```

Reply:
I don't think that would make sense, as this PR is only introducing circuits and redefining the constraints as such. It's not yet using circuits for anything meaningful.

Reply:
I agree. With circuits in place, multivariate polynomials are on their way out. We should postpone revising the architecture until after the transition is complete.